# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/child_thread.h
# Commit: 115187a5f512
# Full Hash: 115187a5f5120c5fda63685fd8a1471339852735
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-26 09:46:54
# Description:
#   Bug 1717993 - Hold a WeakPtr to mExistingListener in NodeChannel, r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D118868
# ==============================================================================

diff -r d0b8984194c6 -r 115187a5f512 ipc/chromium/src/chrome/common/child_thread.h
--- a/ipc/chromium/src/chrome/common/child_thread.h	Fri Jun 25 21:10:50 2021 +0000
+++ b/ipc/chromium/src/chrome/common/child_thread.h	Fri Jun 25 21:30:49 2021 +0000
@@ -16,7 +16,7 @@
 class ResourceDispatcher;
 
 // Child processes's background thread should derive from this class.
-class ChildThread : public IPC::Channel::Listener, public base::Thread {
+class ChildThread : public base::Thread {
  public:
   // Creates the thread.
   explicit ChildThread(Thread::Options options);
@@ -41,10 +41,6 @@
   virtual void CleanUp() override;
 
  private:
-  // IPC::Channel::Listener implementation:
-  virtual void OnMessageReceived(IPC::Message&& msg) override;
-  virtual void OnChannelError() override;
-
   // The message loop used to run tasks on the thread that started this thread.
   MessageLoop* owner_loop_;
 